### PR TITLE
fix(select): select option not cleaned up when binding was updated programmatically

### PR DESCRIPTION
**Pull request by aefox**
### Please read and mark the following check list before creating a pull request:

 - [x] I read and followed the [CONTRIBUTING.md](https://github.com/akveo/nebular/blob/master/CONTRIBUTING.md) guide.
 - [x] I read and followed the [New Feature Checklist](https://github.com/akveo/nebular/blob/master/DEV_DOCS.md#new-feature-checklist) guide.
 
 #### Short description of what this resolves:
This fixes https://github.com/akveo/nebular/issues/1260 where select was not being cleaned up when its binding was set to null programmatically.

### CODE REVIEW

**Comment by aefox** [comment]
Please note that this PR breaks the guideline to ` Use changeDetection: ChangeDetectionStrategy.OnPush where applicable`. See it [here](https://github.com/akveo/nebular/pull/1265/files#diff-6d5a993596400c649c6d5287125c6acaL149).

Because of this I did not write tests since I first need to know this is acceptable.

File: src/framework/theme/components/select/select.component.ts [hunk 1]
```diff
@@ -371,6 +369,7 @@ export class NbSelectComponent<T> implements OnInit, AfterViewInit, AfterContent
 
   writeValue(value: T | T[]): void {
     if (!value) {
+      this.cleanSelection();
```
Comment (yggg) [inline src/framework/theme/components/select/select.component.ts]: Can't we keep `OnPush` and manually call `this.cd.detectChanges` here after selection cleanup?

File: package.json [hunk 2]
```diff
@@ -174,4 +174,4 @@
     "uglifyjs-webpack-plugin": "1.1.5"
   },
   "schematics": "./schematics/dist/collection.json"
-}
\ No newline at end of file
+}
```
Comment (yggg) [inline package.json]: Please, revert whitespace/newline changes here and also in `package-lock.json` and `select-showcase.component.html`.

**Comment by aefox** [comment]
@yggg thanks for the review!

I'm closing this as all changes from https://github.com/akveo/nebular/pull/1282 are also fixing this issue. I've tested all possible scenarios for this ticket and you can see the gist with the local changes [here](https://gist.github.com/aefox/8f3fd6b64d0c8ae3772ba69102e9bb33)

I will update https://github.com/akveo/nebular/pull/1282 to also mention this issue is covered by those changes.

**Pull request closed without merging**